dotcom-rendering/src/web/components/InteractiveContentsBlockComponent.importable.tsx

```diff
@@ -176,7 +176,7 @@ export const InteractiveContentsBlockComponent = ({
 	// -> If we stop intersecting with the current element, and its in view > show the preview element
 	// -> If we're intersecting with an element set that to the active one
 	// -> If the end element is in view, but the last element is not > show the last element
-	useEffect((): any => {
+	useEffect((): void | (() => void) => {
```

Comment:
Do you even need to specify the return type?

```suggestion
	useEffect(function observeElement() {
```

Reply:
It seems if we don't we get a `Not all code paths return a value.` from TS. Guess it doesn't like the ambiguity